src/org/zaproxy/zap/extension/exportreport/Export/ReportExport.java

```diff
@@ -0,0 +1,315 @@
+	
+	private static String entityEncode(String text) throws UnsupportedEncodingException {
+		String result = text;
+
+		if (result == null) {
+			return result;
+		}
+		// There is an encoding issue with the passed in String, this is a fix to maintain encoding and escapes!
+		byte ptext[] = result.getBytes("ISO-8859-1"); 
+		String value = new String(ptext, Utils.utf8); 
+		String temp = XMLStringUtil.escapeControlChrs(value);
+
+		return temp;
+	}
+
+	public static String generateDUMP(String path, String fileName, String reportTitle, String reportBy, String reportFor,
+			String scanDate, String scanVersion, String reportDate, String reportVersion, String reportDesc,
+			ArrayList<String> alertSeverity, ArrayList<String> alertDetails) throws UnsupportedEncodingException {
+
+		SiteMap siteMap = Model.getSingleton().getSession().getSiteTree();
+		SiteNode root = (SiteNode) siteMap.getRoot();
+		int siteNumber = root.getChildCount();
+
+		Report report = new Report();
+		report.setTitle(entityEncode(reportTitle));
+		report.setBy(entityEncode(reportBy));
+		report.setFor(entityEncode(reportFor));
+		report.setScanDate(entityEncode(scanDate));
+		report.setScanVersion(entityEncode(scanVersion));
+		report.setReportDate(entityEncode(reportDate));
+		report.setReportVersion(entityEncode(reportVersion));
+		report.setDesc(entityEncode(reportDesc));
+
+		String description = Constant.messages.getString("exportreport.details.description.label");
+		String solution = Constant.messages.getString("exportreport.details.solution.label");
+		String otherinfo = Constant.messages.getString("exportreport.details.otherinfo.label");
+		String reference = Constant.messages.getString("exportreport.details.reference.label");
+		String cweid = Constant.messages.getString("exportreport.details.cweid.label");
+		String wascid = Constant.messages.getString("exportreport.details.wascid.label");
+		String requestheader = Constant.messages.getString("exportreport.details.requestheader.label");
+		String responseheader = Constant.messages.getString("exportreport.details.responseheader.label");
+		String requestbody = Constant.messages.getString("exportreport.details.requestbody.label");
+		String responsebody = Constant.messages.getString("exportreport.details.responsebody.label");
+
+		try {
+			for (int i = 0; i < siteNumber; i++) {
+				SiteNode site = (SiteNode) root.getChildAt(i);
+				String siteName = ScanPanel.cleanSiteName(site, true);
+				String[] hostAndPort = siteName.split(":");
+				boolean isSSL = (site.getNodeName().startsWith("https"));
+
+				Sites s = new Sites();
+				s.setHost(entityEncode(hostAndPort[0]));
+				s.setName(entityEncode(site.getNodeName()));
+				s.setPort(entityEncode(hostAndPort[1]));
+				s.setSSL(String.valueOf(isSSL));
+
+				List<Alert> alerts = site.getAlerts();
+				Alerts a = new Alerts();
+				String temp = "";
+				for (Alert alert : alerts) {
+					
+					if(!alertSeverity.contains(Alert.MSG_RISK[alert.getRisk()])) {
```

Comment:
This will mismatch with translated messages, the values of Alert.MSG_RISK area always the same (in English), but the values in alertSeverity maybe not.